File: EnMayus/JavaScript/EnMayus.js
/*
 * EnMayuscula
 * Crea una función que reciba un String de cualquier tipo y se encargue de
 * poner en mayúscula la primera letra de cada palabra.
 * - No se pueden utilizar operaciones del lenguaje que lo resuelvan
 * directamente.
 */
function enMayuscula(frase) {
    // Dividir la frase en palabras
    let palabras = frase.split(' ');
  
    // Modificar la primera letra de cada palabra
    let palabrasModificadas = palabras.map(palabra => {
      return palabra.charAt(0).toUpperCase() + palabra.slice(1);
    });
  
    // Unir las palabras modificadas en una cadena
    let resultado = palabrasModificadas.join(' ');
  
    return resultado;
  }
  
  // Preguntar al usuario por una palabra o frase
  let entradaUsuario = prompt("Introduce una palabra o frase: ");
  
  // Aplicar la función
  let resultadoFinal = enMayuscula(entradaUsuario);
  
  // Mostrar el resultado
  console.log("Resultado:", resultadoFinal);
  

File: EnMayus/Python/en_mayus.py
"""
                        EnMayuscula
  Crea una función que reciba un String de cualquier tipo y se encargue de
  poner en mayúscula la primera letra de cada palabra.
  - No se pueden utilizar operaciones del lenguaje que lo resuelvan
  directamente.
"""
def en_mayuscula(frase):
    palabras = frase.split()  # Dividir la frase en palabras
    palabras_modificadas = []

    for palabra in palabras:
        palabra_modificada = palabra[0].upper() + palabra[1:]
        palabras_modificadas.append(palabra_modificada)

    resultado = ' '.join(palabras_modificadas)
    return resultado

# Preguntar al usuario por una palabra o frase
entrada_usuario = input("Introduce una palabra o frase: ")

# Aplicar la función
resultado_final = en_mayuscula(entrada_usuario)

# Mostrar el resultado
print(resultado_final)

